pkg/arvo/neo/cod/std/src/fil/axal.txt

```diff
@@ -0,0 +1,338 @@
+# Axal guide
+
+`+axal` is the recursive tree type that Shrubbery uses to build the namespace.
+
+For userspace developers, axals are useful when you're writing `/con` files for your frontend and you want to work with the descendants of whichever shrub you're viewing in the frontend. (e.g. You're rendering a todo item and you want to render its sub-tasks, which are nodes beneath the main task in the namespace.)
+
+We work with axals using the `+of` core defined in `/sur/neo.hoon`, whose arms are documented below.
+
+[note on piths and paths, how `of:neo` is used and how to work with piths and paths, convert from path to pith, etc.]
+
+The `+axal` type itself looks like this.
+
+```
+++  axal
+  |$  [item]
+  [fil=(unit item) kid=(map iota $)]
+```
+
+One way to think of the Shrubbery namespace is as a filesystem, where files could be data, code, or something in between. So the `+axal` represents a directory of such files. The `fil` is the state of the file, which may or may not be null. The `kid` map stores everything below this file in the directory. The map's keys are the path segments immediately below this layer of the directory, and its values are the subtrees that correspond to those paths.
+
+***
+
+## `+anc`
+
+[??? - i see what this is doing just dk why]
```

Comment:
Gathers a list of *populated* ancestors. i.e. if you have a tree with stuff at /foo /foo/bar/baz, but not at /foo/bar, then doing `(~(anc of axal) /foo/bar/baz/qux)` it would return the list `~[/foo /foo/bar/baz]`. Most of these are not useful for the average end user tbh.